modules/gateway/peers.go

```diff
@@ -493,7 +493,7 @@ func (g *Gateway) managedConnect(addr modules.NetAddress) error {
 			}
 			defer g.threads.Done()
 
-			err := g.RPC(addr, name, fn)
+			err := g.managedRPC(addr, name, fn)
```

Comment:
definitely a potential deadlock - right above the call the thread calls `g.threads.Add()`, and inside of the blocking call to `g.RPC` is another call to `g.threads.Add()` - deadlock!